halo1/model.py

```diff
@@ -0,0 +1,76 @@
+import bpy
+
+from reclaimer.hek.defs.mode import mode_def
+from reclaimer.hek.defs.mod2 import mod2_def
+from reclaimer.model.jms import read_jms
+from reclaimer.model.model_decompilation import extract_model
+
+from ..scene.shapes import create_sphere
+
+def read_halo1model(filepath):
+	'''Takes a halo1 model file and turns it into a jms object.'''
+
+	# TODO: Use a tag handler to see if these files actually are what they
+	# say they are. We can get really nasty parsing problems if they aren't.
+
+	# These two model types can be imported the same way because of their
+	# nearly matching structures when built into a python object.
+	if (filepath.lower().endswith('.gbxmodel')
+	or filepath.lower().endswith('.model')):
+		# Load model
+		tag = None
+		if filepath.lower().endswith('.gbxmodel'):
+			tag = mod2_def.build(filepath=filepath)
+		else:
+			tag = mode_def.build(filepath=filepath)
+		#TODO: Get all lod permutations.
+		#Only getting the superhigh perms for now
+		jms = extract_model(tag.data.tagdata, write_jms=False)[0]
+		return jms
+
+	if filepath.lower().endswith('.jms'):
+		# Read jms file into string.
+		jms_string = ""
+		with open(filepath, 'r') as jms_file:
+			jms_string = jms_file.read()
+		# Read Jms data from string.
+		jms = read_jms(jms_string)
+		# Make sure it's a Halo 1 jms
+		if jms.version != "8200":
+			raise ValueError('Not a Halo 1 jms!')
+
+		return jms
+
+def import_halo1_nodes(jms, scale=1.0, node_size=0.02):
+	'''
+	Import all the nodes from a jms into the scene and returns a list of them.
+	'''
+	# Jms are scaled at x100 halo. Undo that.
+	scale = scale/100
```

Comment:
You defined a constant to deal with this exact conversion. You should use that here. Better yet, you could make a small utility file with functions like `scale_max_to_halo` or something

Reply:
I think I'll condense the two scalings into one since we're using the JmsModel class which has jms scaling.